modules/db/src/main/java/org/mule/module/db/internal/domain/param/DefaultInputQueryParam.java

```diff
@@ -30,4 +32,18 @@ public Object getValue()
     {
         return value;
     }
+
+    /**
+     * @return true if the InputQueryParam references a param that must be defined through a <db:in-param> element in the configuration file.
+     * false if the InputQueryParam is defined through a literal or a MEL expression
+     */
+    public boolean isDbInParam()
+    {
+        return isDbParam;
+    }
+
+    public void setDbParam(boolean dbParam)
```

Comment:
This class was supposed to be immutable. Find a different way to create it with the right value